engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -226,6 +234,31 @@ private[spel] class Typer(classLoader: ClassLoader) extends LazyLogging {
     }
   }
 
+  private def checkEqualityLikeOperation(validationContext: ValidationContext, node: Operator, current: TypingContext): ValidatedNel[ExpressionParseError, CollectedTypingResult] = {
+    typeChildren(validationContext, node, current) {
+      case left :: right :: Nil if commonSupertypeFinder.commonSupertype(right, left)(NumberTypesPromotionStrategy.ToSupertype) != Typed.empty => Valid(Typed[Boolean])
+      case left :: right :: Nil => invalid(s"Invalid operands: $left ${node.getOperatorName} $right")
```

Comment:
can we think of some better error message? So that user knows it's about having not really comparable left and right?